testing/deployer/sprawl/catalog.go

```diff
@@ -277,6 +444,9 @@ func (s *Sprawl) registerCatalogSidecarService(
 	if svc.DisableServiceMesh {
 		panic("not valid")
 	}
+	if node.IsV2() {
+		panic("don't call this")
```

Comment:
under what circumstances would this get called with a v2 node?

Reply:
It would not. This is a precondition "you're holding it wrong" panic, like the others in similar methods in this file.